### DISCUSSION TITLE
Transcripts always misses the first words (at least in french)

### DISCUSSION THREAD

**Discussion by yafkari** [deepgram/community#546] [General help]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

We have an issue with our service that sends a mp3 file to be transcribe with `await deepgram.listen.prerecorded.transcribeFile(Buffer.from(contents, "base64"), options);`.

Everytime we send a file to be transcribed, the first word is missing.

Most of our tests were using french language so I cannot say 100% that this is not the case in eng but from our experience with Deepgram, the first word of the transcription is always missing and it causes issues with our testers that always have to correct their transcriptions by adding the missing first word. In the worst case, it can also make the the next word wrongly interpreted.

I didn't find anyone else experimenting this or complaining about it so I suppose it's related to the french language? I would also be curious to know if this also happens with other languages.

If I have the time I would also like to add a 1s of blank in the audio file to see if it can help detect the first word.

EDIT: 

Just found a transcription where the second word start time was "0.08s" so i suppose the first word was too close to the beginning of the audio.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

/v1/listen?punctuate=true&detect_language=true&paragraphs=false&model=nova-2-general&diarize=false&smart_format=true&numbers=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

4213aae1-ae72-499b-bff9-3a66734177f7

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by netw0rkf10w** [comment]
Happened to me as well! It seems that the support for French language is still very poor :(

**Comment by davidvonthenen** [comment]
hi @netw0rkf10w and @yafkari 

Sorry, we missed this question. If you are customer with a support contract, I would recommend submitting a ticket through your customer portal for quicker response. PAYG support is community support, but we usually try to acknowledge new community issues. It seems like this one fell through the cracks.

There was a refresh to all the models recently (especially around non-english languages) on Monday I believe. Please give it a try and let me know if you are still running into the same issue.

**Reply by yafkari** [reply]
Hey @dvonthenen no problem, i get it ^^

Just tried and I still get the issue with an audio sadly. It found the second word at 0.16s but didn't understand the first one.

**Reply by davidvonthenen** [reply]
Do you know what kind of a delay there is between connection to the websocket and sending the first audio payload? It would be interesting to get timing on that.

**Reply by yafkari** [reply]
hey @dvonthenen sorry for the delay, quite a busy month for me ^^'

Don't know if that's what you were looking for but here is what i can provide you. (Obviously it's timings on my local machine and not on the server)
![image](https://github.com/user-attachments/assets/1b0bbdf3-f0b5-40be-aaf5-0e1c00f5e527)

I took the opportunity to do some tests and seems like the base modal was able to get the first word.

![image](https://github.com/user-attachments/assets/1070ecef-f38b-491b-9335-73cd2cc7933b)

The enhanced model also wasn't able to get the first word
![image](https://github.com/user-attachments/assets/110aa2cb-a0e0-4c09-a7b9-232d6f5b3e55)


In the audio i used, it starts in french with "Si tu es fan de", meaning "if you are a fan of" and as I already mentionned it's really the "Si" that the models are not getting here.

I don't recall totally about the other cases but that's not the only word, I really think it is when it starts too quickly. Unfortunately if you miss one word, you can't split the text in groups accurately which is a blocker for me :(

**Reply by davidvonthenen** [reply]
hi @yafkari 

No worries about the delay. Trust me... I know that busy feeling. 🤣

Did you try the `nova-2` model?

Also, I just realized that `detect_language` might be a pretty significant hit when first starting vs knowing what languages the calls are in and just setting it. Another metric that could be of interest is trying this out and seeing what the difference in time is.

**Reply by yafkari** [reply]
@dvonthenen 

The first screenshot was actually the nova-2 model. (The initial request id in my first message is too)

I couldn't test on nova modal since it doesn't support french.

![image](https://github.com/user-attachments/assets/bdf0b54e-e9ce-4f05-9194-1699d199290a)

Did some runs (15-20) with language or detect_language, i don't see any significant change. But whatever the time it takes apparently it doesn't get it.

Here are some duration from my tests

```
model: base
with language parameter
4.67s
1.58s
2.17s
1.3s
0.8s
1.3s

with detect_language
4.5s
4.2s
2.4s
3.3s
1.7s
1.3s
1.3s

model: nova-2
with language parameter
1.5
1
4
2.22
1.4

with detect_language
1.2
1.3
1.5
1.2
```

I was expecting it would be the same but for the science i did the tests with the rest api while i use the js sdk on the server but as i expected, it's the same.

PS: just realised I'm killing my free credits but it's worth it 😆

**Reply by davidvonthenen** [reply]
I will take this back to the team. The piece about the REST call doing the same thing is very valuable.

**Reply by yafkari** [reply]
Hey @dvonthenen, just tried with Whisper cloud model from the Deepgram playground page and it actually detected the first word!

**Reply by yafkari** [reply]
Me again @dvonthenen , as a workaround I add one second of blank at the beginning of the audio and fix the timings later by myself.

(I also tried to add only 0.1s but it was still not getting the first word, so to be safe i'll add one second and fix the timings later programmatically)

Still a pain, but at least i can continue using nova-2
